cmd/kubeadm/app/phases/upgrade/staticpods.go

```diff
@@ -192,3 +281,19 @@ func rollbackOldManifests(oldManifests map[string]string, origErr error, pathMgr
 	// Let the user know there we're problems, but we tried to reçover
 	return fmt.Errorf("couldn't upgrade control plane. kubeadm has tried to recover everything into the earlier state. Errors faced: %v", errs)
 }
+
+// rollbackEtcdData rolls back the the content of etcd folder if something went wrong
+func rollbackEtcdData(cfg *kubeadmapi.MasterConfiguration, origErr error, pathMgr StaticPodPathManager) error {
```

Comment:
In this func, I want to:
 - Move back the old manifest
 - Check if the old etcd version comes back up again
 - If it does: Say to the user we tried to upgrade but couldn't, but restored everything
 - If it doesn't: Say to the user we tried to upgrade but couldn't, and couldn't restore the earlier state. Tell the user the data was backed up to {backup-dir} instead of overwriting any data that might or might not be in use

Reply:
Here is a comment from @jamiehannaford : '''
If the etcd upgrade goes wrong:

We should replace the current data dir (if any) with restore old one
Restore manifests
Check the restored etcd works. If it's not responding, tell the user something like "Hey something went wrong. We tried restoring but that failed. Here is the location to your backed up data"
'''
Please sync up with him what would be the final solution. As restoring etcd data was suggested by him. (Unless of course I misread his comment).

Reply:
I'll sync up with him.
I think I disagree on `We should replace the current data dir (if any) with restore old one` => instead defer this to the user. We tried our best and did back up, but won't start overriding any data just like that to be safe.